sklearn/metrics/_pairwise_distances_reduction/meson.build

```diff
@@ -39,7 +39,7 @@ _datasets_pair_pyx = custom_target(
 _datasets_pair = py.extension_module(
   '_datasets_pair',
   _datasets_pair_pyx,
-  dependencies: [np_dep, openmp_dep],
+  dependencies: [np_dep],
```

Comment:
curious why you thought these are needed before then. And are they really not needed?

Reply:
> curious why you thought these are needed before then.

This is the most complicated `meson.build` there is and I would not be surprised if I cut some corners when I put it together.

> And are they really not needed?

I am pretty sure they are not needed since there is nothing related to OpenMP in https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/metrics/_pairwise_distances_reduction/_datasets_pair.pyx.tp.